test/kt/esimd_radix_sort_by_key.cpp

```diff
@@ -0,0 +1,128 @@
+                                 ", isAscending: " +  std::to_string(isAscending) +
+                                 ", RadixBits: " + std::to_string(RadixBits);
+    std::string msg = "wrong results with oneapi::dpl::begin/end (keys)" + parameters_msg;
+    EXPECT_EQ_N(expected_keys.begin(), actual_keys.begin(), size, msg.c_str());
+    msg = "wrong results with oneapi::dpl::begin/end (values)" + parameters_msg;
+    EXPECT_EQ_N(expected_values.begin(), actual_values.begin(), size, msg.c_str());
+}
+
+template<typename KeyT, typename ValueT, bool isAscending, std::uint32_t RadixBits, sycl::usm::alloc _alloc_type, typename KernelParam>
+void test_usm(std::size_t size, KernelParam param)
+{
+    sycl::queue q = TestUtils::get_test_queue();
+
+    std::vector<KeyT> expected_keys(size);
+    std::vector<ValueT> expected_values(size);
+    generate_data(expected_keys.data(), size, 6);
+    generate_data(expected_values.data(), size, 7);
+
+    TestUtils::usm_data_transfer<_alloc_type, KeyT> keys(q, expected_keys.begin(), expected_keys.end());
+    TestUtils::usm_data_transfer<_alloc_type, ValueT> values(q, expected_values.begin(), expected_values.end());
+
+    auto expected_first  = oneapi::dpl::make_zip_iterator(std::begin(expected_keys), std::begin(expected_values));
+    std::stable_sort(expected_first, expected_first + size, CompareKey<isAscending>{});
+
+    oneapi::dpl::experimental::kt::esimd::radix_sort_by_key<isAscending, RadixBits>(
+        q, keys.get_data(), keys.get_data() + size, values.get_data(), param).wait();
+
+    std::vector<KeyT> actual_keys(size);
+    std::vector<ValueT> actual_values(size);
+    keys.retrieve_data(actual_keys.begin());
+    values.retrieve_data(actual_values.begin());
+
+    std::string parameters_msg = ", n: " + std::to_string(size) +
+                                 ", sizeof(key): " + std::to_string(sizeof(KeyT)) +
+                                 ", sizeof(value): " + std::to_string(sizeof(ValueT)) +
+                                 ", isAscending: " +  std::to_string(isAscending) +
+                                 ", RadixBits: " + std::to_string(RadixBits);
+    std::string msg = "wrong results with USM (keys)" + parameters_msg;
+    EXPECT_EQ_N(expected_keys.begin(), actual_keys.begin(), size, msg.c_str());
+    msg = "wrong results with USM (values)" + parameters_msg;
+    EXPECT_EQ_N(expected_values.begin(), actual_values.begin(), size, msg.c_str());
+}
+
+template <std::uint16_t DataPerWorkItem, std::uint16_t WorkGroupSize>
+using param_type = oneapi::dpl::experimental::kt::kernel_param<DataPerWorkItem, WorkGroupSize>;
+
+int main()
+{
+    const std::vector<std::size_t> sizes = {
+        6, 16, 43, 256, 316, 2048, 5072, 8192, 14001, 1<<14,
+        (1<<14)+1, 50000, 67543, 100'000, 1<<17, 179'581, 250'000, 1<<18,
+        (1<<18)+1, 500'000, 888'235, 1'000'000, 1<<20, 10'000'000
+    };
+
+    for(auto size: sizes)
```

Comment:
I believe better to organize this test like esimd_radix_sort.cpp and specify all test params from CMakeLists.txt file through CMake params.
Otherwise we increate build time for this test and may don't understand which combination of test params exactly failed.

Reply:
It's a good idea in long-term, but it's a pretty big chunk of work. Would you mind if we do it in a some follow-up PR? 

The current `esimd_radix_sort_by_key` test builds in ~3-5 minutes and it covers the most basic cases (I believe so). It essentially proves that key-value sort works.